Adding namespace support documentation

ofiriro3 opened the pull request:
## What does this PR do?
This PR is adding a paragraph in our integration README, that the cloud security posture plugin support installation only using the `default` namespace.
 

### What does this PR include?
Only readme changes.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).



## Related issues
- Relates https://github.com/elastic/security-team/issues/5831


## Screenshots
<img width="967" alt="image" src="https://user-images.githubusercontent.com/19221736/217196148-57c82818-4a7f-4281-9e91-00c1bbfe2689.png">

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-5197/detail/PR-5197/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-5197/detail/PR-5197/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-5197/detail/PR-5197/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-5197/detail/PR-5197/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_5197.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2023-02-07T08:38:32.485Z&rangeTo=2023-02-07T08:58:32.485Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=b7ef685889a4ffa3c15f5a076336b3b2&transactionId=f89080efb1e7f6af)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-02-07T08:48:32.485+0000


* Duration: 15 min 43 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 2  |
| Skipped      | 0 |
| Total        | 2   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`0/0`) | :green_heart: 
Files | 100.0% (`0/0`) | :green_heart: 
Classes | 100.0% (`0/0`) | :green_heart: 
Methods | 25.0% (`1/4`) | :grey_exclamation: 
Lines | 100.0% (`0/0`) | :green_heart: 
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

oren-zohar commented on packages/cloud_security_posture/docs/README.md:
```diff
@@ -40,7 +40,12 @@ This integration does not currently support the security posture assessment of t
 The integration supports **elastic agent** version 8.5 and above.
 
-## Integration Requirments 
+## Namespace support
+The KSPM installation is currently only supported using the default namespace.
+
+Failing to install the integration using the default namespace will result in a invalid integration state. 
```
suggestion:
```suggestion
## Namespace support
KSPM is currently only supported with the default namespace configuration.

Failing to install the integration using the default namespace will result in an invalid integration state. 
```

kfirpeled reviewed (commented):
I believe you are missing a new entry in the changelog for making this right
looking at https://github.com/elastic/integrations/blob/backport-aws-7.15.0/packages/aws/changelog.yml it seems we are missing a step here

ofiriro3 commented:
> I believe you are missing a new entry in the changelog for making this right
looking at https://github.com/elastic/integrations/blob/backport-aws-7.15.0/packages/aws/changelog.yml it seems we are missing a step here

I didn't add a change log on purpose, I'll try to explain here:
Since we are "amending" the last commit(1.0.5) 
```script
  - version: "1.0.5"
  changes:
    - description: Documentation bugfix
      type: bugfix
      link: https://github.com/elastic/integrations/pull/4504
```
Looking at our current [changelog](https://github.com/elastic/integrations/blob/main/packages/cloud_security_posture/changelog.yml), I see that we already have a commit for 1.0.6, meaning that the backport will result in conflicts and probably won't succeed and will cause unexpected behavior.
I looked at the integration repository and I saw other commits with no change logs, that were there to "amend" the last commit.

If you still think that we should add the commit log, so I think that we will need to think of a different strategy here.
@kfirpeled

ofiriro3 commented:
> I believe you are missing a new entry in the changelog for making this right
looking at https://github.com/elastic/integrations/blob/backport-aws-7.15.0/packages/aws/changelog.yml it seems we are missing a step here

I added the changelog but I wasn't sure if I should add it.
I will try to explain:
As far as I know, if there is no commit log so no package will be published - so if looking at this alone I should have added the Changelog.
But since this is a bugfix flow, I assumed that the package is being published regardless based on the branch name.

Anyway, I added the changelog, and I added a number that won't collide with the next changelog we have.
Back at the time, we haven't aligned our version to Kibana version so 1.0.6 is actually referring to 8.6 Kibana version

jsoriano commented on packages/cloud_security_posture/changelog.yml:
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.0.9"
```
Shouldn't it be 1.0.6?

ofiriro3 replied:
We just recently started to align the Kibana version with our change log versioning.
So the convention is
1.1.* - 8.6 Kibana
1.2.* - 8.7 Kibana.

But on 1.0.* some commits refer to 8.5 Kibana and some refer to 8.6 Kibana.
And "1.0.6" commit is already in use in 8.6 Kibana, actually, it's the first 8.6 Kibana.

The next 1.0.* that was free was "1.0.9".
https://github.com/elastic/integrations/blob/main/packages/cloud_security_posture/changelog.yml

I just want to mention here that we have auto-upgrading integration, which will always install the latest integration available for the Kibana version.

jsoriano replied:
Ah ok, thanks, sounds good then.

jsoriano commented:
```
[2023-02-08T09:13:26.428Z] Error: checking package failed: linting package failed: found 1 validation error:
[2023-02-08T09:13:26.428Z]    1. current manifest version doesn't have changelog entry
```
Ah, you also need to update the version in the manifest.

ofiriro3 commented:
I added the new changelog, and run all the tests. 

Screenshot attached:
<img width="1525" alt="image" src="https://user-images.githubusercontent.com/19221736/217503858-ccb8b9cf-c3c8-4276-8e7e-dce34f5c5beb.png">

Ready for review

ofiriro3 commented:
I am closing this PR.

I'd like to stick with the convention of "major.minor" from the documentation so it would be easier to track and backport next time
